### PR TITLE
Branch to mergeto develop

### PR DESCRIPTION
**Pull request by DanielLaga84**
Please check our newest version of fully working application.

### CODE REVIEW

File: src/main/java/com/infoshare/drinkywinky/Main.java [hunk 1]
```diff
@@ -8,7 +8,7 @@
     public static void main(String[] args) {
 
         /*
-         display main menu
+         display main menu11
```
Comment (RozwalkaWojciech) [inline src/main/java/com/infoshare/drinkywinky/Main.java:11]: Why are you change this line? What's the difference between menu and menu11??

File: src/main/java/com/infoshare/drinkywinky/menu/PropertiesMenu.java [hunk 2]
```diff
@@ -15,12 +14,13 @@
...
 
 import static com.infoshare.drinkywinky.menu.Menu.SCANNER;
+import static com.infoshare.drinkywinky.menu.Menu.WrongNumber;
 import static com.infoshare.drinkywinky.properties.AppConfig.dateFormat;
 import static com.infoshare.drinkywinky.properties.AppConfig.recipeSortType;
 
 
 public class PropertiesMenu {
-    private static Object SORT_TYPE = AppConfig.recipeSortType;
+    public static Object SORT_TYPE = AppConfig.recipeSortType;
```
Comment (RozwalkaWojciech) [inline src/main/java/com/infoshare/drinkywinky/menu/PropertiesMenu.java:23]: Uppercase should be in static final fields

File: src/main/java/com/infoshare/drinkywinky/menu/Menu.java [hunk 3]
```diff
@@ -67,17 +71,21 @@ public static void mainMenu() {
...
-                    STDOUT.info("│               TRY AGAIN!                 │\n");
-                    STDOUT.info("│                                          │\n");
-                    STDOUT.info("└──────────────────────────────────────────┘\n");
-                    STDOUT.info("\n");
+                    WrongNumber();
                     break;
             }
         }
     }
+
```
Comment (RozwalkaWojciech) [inline src/main/java/com/infoshare/drinkywinky/menu/Menu.java:79]: Comment, how you can use this method at another display menu

File: src/main/java/com/infoshare/drinkywinky/menu/SubmenuCreator.java [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+            countNumberOfMenuPages();
+            drawSubmenuContent();
+        }
+        if (SORT_TYPE.equals("ASC")) {
+            listOfSubmenuElements.stream().sorted().collect(Collectors.toList());
+            listOfSubmenuElements.sort(Comparator.naturalOrder());
+            Repository.saveToFile(Repository.getInstance().getDrinkList(), USER_DATA_BASE_PATH_NAME);
+            countNumberOfMenuPages();
+            drawSubmenuContent();
+
```
Comment (RozwalkaWojciech) [inline src/main/java/com/infoshare/drinkywinky/menu/SubmenuCreator.java:79]: delete this line

File: src/main/java/com/infoshare/drinkywinky/menu/SubmenuCreator.java [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+        }
+
```
Comment (RozwalkaWojciech) [inline src/main/java/com/infoshare/drinkywinky/menu/SubmenuCreator.java:81]: delete this line

File: src/main/java/com/infoshare/drinkywinky/menu/SubmenuCreator.java [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+            STDOUT.info("│   \u001b[33mP\u001b[0m <- Previous page    Next page -> \u001b[33mN\u001b[0m   │\n");
+        }
+        if (pageNumber == numberOfPages - 1 && numberOfPages > 1) {
+            STDOUT.info("│   \u001b[33mP\u001b[0m <- Previous page                     │\n");
+        }
+        if (pageNumber == 0 && numberOfPages > 1) {
+            STDOUT.info("│                         Next page -> \u001b[33mN\u001b[0m   │\n");
+        }
+    }
+
```
Comment (RozwalkaWojciech) [inline src/main/java/com/infoshare/drinkywinky/menu/SubmenuCreator.java:120]: delete this line

File: src/main/java/com/infoshare/drinkywinky/menu/SubmenuCreator.java [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+        if (userChoiceString.equalsIgnoreCase("X"))
+            Menu.mainMenu();
+    }
+
+    protected void chooseSpecificSubmenuElement() {
+        /**
+         * @param listSelectedByName List of ALL {@class Drink} objects isolated from updated database by the NAME pointed by user
+         * @param setSelectedByCategory Set of ALL {@class Drink} objects isolated from updated database by the CATEGORY pointed by user
+         * @param listSelectedByIngredients List of ALL {@class Drink} objects isolated from updated database by the INGREDIENT pointed by user
+         *
```
Comment (RozwalkaWojciech) [inline src/main/java/com/infoshare/drinkywinky/menu/SubmenuCreator.java:183]: delete this line

File: src/main/java/com/infoshare/drinkywinky/repositories/Repository.java [hunk 8]
```diff
@@ -70,7 +85,7 @@ public void remove(Drink drink) {
 
     public void addFavorite(Drink drink) {
         favoriteDrinkList.addDrink(drink);
-        ((DateTimeFormatter) DATE_FORMAT).format(LocalDateTime.now());
+        /*((DateTimeFormatter) DATE_FORMAT).format(LocalDateTime.now());*/
```
Comment (radoslaw-piwowarczyk) [inline src/main/java/com/infoshare/drinkywinky/repositories/Repository.java:88]: remove this

File: src/main/java/com/infoshare/drinkywinky/menu/favourite/ManageFavourite.java [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+public class ManageFavourite {
+    private static DrinkList drinkList;
+    private static final Scanner SCANNER = new Scanner(System.in);
+
+    private static final Logger STDOUT = LoggerFactory.getLogger("CONSOLE_OUT");
+
+    public static void addRemoveRecipe() {
+
+
```
Comment (RozwalkaWojciech) [inline src/main/java/com/infoshare/drinkywinky/menu/favourite/ManageFavourite.java:23]: delete this line

File: src/main/java/com/infoshare/drinkywinky/menu/favourite/ManageFavourite.java [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+        int mainExitCode = 0;
+
+        while (mainExitCode != 3) {
+
+
```
Comment (RozwalkaWojciech) [inline src/main/java/com/infoshare/drinkywinky/menu/favourite/ManageFavourite.java:28]: delete this line

File: src/main/java/com/infoshare/drinkywinky/menu/favourite/ManageFavourite.java [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+                    break;
+                case 4:
+                    STDOUT.info(" CHOSEN : 4. Back to menu  \n");
+                    Menu.mainMenu();
+                    mainExitCode = 3;
+                    break;
+                default:
+                    WrongNumber();
+                    break;
+
```
Comment (RozwalkaWojciech) [inline src/main/java/com/infoshare/drinkywinky/menu/favourite/ManageFavourite.java:63]: delete this line

File: src/main/java/com/infoshare/drinkywinky/menu/favourite/ManageFavourite.java [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+            }
+        }
+
```
Comment (RozwalkaWojciech) [inline src/main/java/com/infoshare/drinkywinky/menu/favourite/ManageFavourite.java:66]: delete this line

File: src/main/java/com/infoshare/drinkywinky/model/Drink.java [hunk 13]
```diff
@@ -127,10 +126,10 @@ public String getGlass() {
 
     @Override
     public String toString() {
```
Comment (RozwalkaWojciech) [inline src/main/java/com/infoshare/drinkywinky/model/Drink.java:128]: Coloring  display menu should be in Utils

File: src/main/java/com/infoshare/drinkywinky/utils/Utils.java [hunk 14]
```diff
@@ -4,11 +4,7 @@
 import com.infoshare.drinkywinky.model.DrinkList;
 
 import java.math.BigInteger;
-import java.util.ArrayList;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
-import java.util.UUID;
+import java.util.*;
```
Comment (RozwalkaWojciech) [inline src/main/java/com/infoshare/drinkywinky/utils/Utils.java:7]: No no no!!!! 
Please do not replace imports with one general!!!

File: src/main/resources/drink list.json [hunk 15]
```diff
@@ -1 +1 @@
...
...d Watermelon Margarita","category":"Cocktail","recipe":"In a mason jar muddle the watermelon and 5 mint leaves together into a puree and strain. Next add the grapefruit juice, juice of half a lime and the tequila as well as some ice. Put a lid on the jar and shake. Pour into a glass and add more ice. Garnish with fresh mint and a small slice of watermelon.","alcoholic":"Alcoholic","glass":"Collins glass","ingredients":["Mint","Watermelon","Tequila","Lime","Grapefruit Juice","Mint","Watermelon"],"measures":["Garnish with","Garnish with","1 shot","Juice of 1/2","1/3 Cup","5","1/2 cup"],"fakeIngredient":null,"fakeMeasure":null},{"dateModified":"10-08-2020 19:38:51","id":"1702587","name":"KakiTest","category":"Normal","recipe":"Blabla","alcoholic":"No-Alcoholic","glass":"Round","ingredients":["Arshenik"],"measures":["1mg"],"fakeIngredient":null,"fakeMeasure":null},{"dateModified":"10-08-2020 19:38:51","id":"0887657","name":"MALIBU","category":"Coctail","recipe":"Mix all ingredients together. First Malibu, than with milk and ice.","alcoholic":"Alcoholic","glass":"High glass","ingredients":["Malibu","Milk","Ice"],"measures":["200ml","200ml",""],"fakeIngredient":null,"fakeMeasure":null}]}
```
Comment (RozwalkaWojciech) [inline src/main/resources/drink list.json:1]: Please add always origin jason data base, without test

File: src/main/resources/favorite drink list.json [hunk 16]
```diff
@@ -0,0 +1 @@
+{"allDrink":[{"dateModified":"19:22:31 08-10-2020","id":"17216","name":"Tommy's Margarita","category":"Ordinary Drink","recipe":"Shake and strain into a chilled cocktail glass.","alcoholic":"Alcoholic","glass":"Old-Fashioned glass","ingredients":["Agave syrup","Lime Juice","Tequila"],"measures":["2 spoons","1.5 cl","4.5 cl"],"fakeIngredient":null,"fakeMeasure":null},{"dateModified":"19:22:31 08-10-2020","id":"1702587","name":"KakiTest","category":"Normal","recipe":"Blabla","alcoholic":"No-Alcoholic","glass":"Round","ingredients":["Arshenik"],"measures":["1mg"],"fakeIngredient":null,"fakeMeasure":null},{"dateModified":"19:22:31 08-10-2020","id":"0887657","name":"MALIBU","category":"Coctail","recipe":"Mix all ingredients together. First Malibu, than with milk and ice.","alcoholic":"Alcoholic","glass":"High glass","ingredients":["Malibu","Milk","Ice"],"measures":["200ml","200ml",""],"fakeIngredient":null,"fakeMeasure":null}]}
```
Comment (RozwalkaWojciech) [inline src/main/resources/favorite drink list.json:1]: List should be empty

**Review by RozwalkaWojciech — APPROVED** [review]
Good job, fix the details

**Pull request merged**